experimental/dds/tree2/src/test/feature-libraries/editable-tree-2/objectFactory.spec.ts

```diff
@@ -196,6 +196,49 @@ describe("SharedTreeObject factories", () => {
 		});
 	});
 
+	it("produce proxies that are hydrated before the tree can be read", () => {
+		// This regression test ensures that proxies can be produced by reading the tree during change events.
+		// Previously, this was not handled correctly because proxies would not be hydrated until after all change
+		// events fired. If a user read the tree during a change event and produced a proxy, that proxy would not
+		// be the same as the one that is about to be hydrated for the same underlying edit node, and thus hydration
+		// would fail because it tried to map an edit node which already had a proxy to a different proxy.
+		// TODO: remove any cast when `viewWithContent` is properly typed with proxy types
+		const view = viewWithContent({ schema, initialTree: initialTree as any });
+		const log: [number, number, number][] = [];
+		function logData() {
+			const objectContent = view.root.child.content;
+			const listContent = view.root.grand.child.list[view.root.grand.child.list.length - 1];
+			assert(listContent !== undefined);
+			const mapContent = view.root.grand.child.map.get("a");
+			assert(mapContent !== undefined);
+			log.push([objectContent, listContent.content, mapContent.content]);
+		}
+		Tree.on(view.root, "beforeChange", () => {
+			logData();
```

Comment:
You could pass in the name of the event source and include it in the log to make it more self-documenting.
```suggestion
			logData("beforeChange");
```
You could also push a string in the log whenever you make a edit (or check the log after each edit and clear it) so the ordering between edits and events is also self-documenting.

None of this is really needed.

Reply:
Those are good ideas, thank you!

Reply:
To your point that "none of this is really needed" - you're right. I just got rid of the logs entirely. These tests don't care what the data is, they just care that it was read. I was bothered by the TS error "you can't just read a property, you have to _do_ something with it", and got carried away. I've settled to just check that the results of the reads are defined :)